### PR TITLE
Adopt AIP-0203: Field Behavior

### LINKED ISSUE #29: Adopt AIP-0203 Field behavior documentation

### PR DESCRIPTION
**Pull request by rambleraptor**
Fixes #29 

Proto changes in https://github.com/aep-dev/aep/pull/13

It's unclear to me if we need an OpenAPI extension, but I'd guess we do?

## 🍱 Types of changes

What types of changes does your code introduce to AEP? _Put an `x` in the boxes
that apply_

- [ ] Enhancement
- [ ] [New proposal](https://aep.dev/1#workflow)
- [X] Migrated from google.aip.dev
- [ ] Chore / Quick Fix

## 📋 Your checklist for this pull request

Please review the [AEP Style and Guidance](https://aep.dev/style-guide) for
contributing to this repository.

### General

- [X] Basic [Guidance](https://aep.dev/style-guide#guidance) is met.
- [X] Ensure that your PR
      [references AEPs](https://aep.dev/style-guide#referencing-aeps)
      correctly.
- [X] [My code has been formatted](https://aep.dev/contributing#formatting)
      (usually `prettier -w .`)

### Additional checklist for a new AEP

- [ ] A new AEP **should** be no more than two pages if printed out.
- [ ] Ensure that the PR is editable by maintainers.
- [ ] Ensure that [File structure](https://aep.dev/style-guide#file-structure)
      guidelines are met.
- [ ] Ensure that
      [Document structure](https://aep.dev/style-guide#document-structure)
      guidelines are met.

💝 Thank you!

### CODE REVIEW

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [inline aep/general/0203/aep.md.j2:151]: nit: I think this type of enumeration should belong in the standard methods - it's a big backwards to do this type of cross-referencing about a specific standard method back to a lower-level AEP around behavior.

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [inline aep/general/0203/aep.md.j2:161]: I don't think this section makes sense - we definitely consider empty lists, zeros, and empty messages as valid inputs for a required fields. 

How about just removing this section?

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [inline aep/general/0203/aep.md.j2:174]: this note further emphasizes to me that we need to remove this part of the guidance.

Why have a required boolean, if the value for a required boolean must always be true?

proto3 the optional keyword to differentiate between unset and set values.

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [inline aep/general/0203/aep.md.j2:189]: is this sentence necessary? it seems pretty clear what this annotation guarantees based on the last two paragraphs.

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [inline aep/general/0203/aep.md.j2:211]:
```suggestion
- Adding `IDENTIFIER` to an existing `path` field
```

we use path now for AEPs.

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [inline aep/general/0203/aep.md.j2:263]: we can get rid of this, since AEPs don't have a strong precedent yet?

**Review by toumorokoshi — CHANGES_REQUESTED** [review]
Thanks! A few notes on this one - really on issues in the existing issues.

File: aep/general/0203/aep.md.j2
Comment (rambleraptor) [reply aep/general/0203/aep.md.j2:189]: Yeah, I can remove this.

File: aep/general/0203/aep.md.j2
Comment (rambleraptor) [reply aep/general/0203/aep.md.j2:174]: If we're only supporting proto3, I agree with this.

File: aep/general/0203/aep.md.j2
Comment (rambleraptor) [reply aep/general/0203/aep.md.j2:151]: I agree with this. I'll make an issue for this.

File: aep/general/0203/aep.md.j2
Comment (rambleraptor) [reply aep/general/0203/aep.md.j2:57]: Yeah, agree on this.

Comment (toumorokoshi) [reply aep/general/0203/aep.md.j2:57]: cool, I see a couple options:

1. wait until Friday to discuss this idea in depth and potentially fix it in this PR.
2. file an issue and we can pick it up later.

Up to you!

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0203/aep.md.j2:165]: In proto2 - what value would a boolean that is always set to true serve? Would the idea be that it's a flag that helps others understand that a new version of the schema is used?

File: aep/general/0203/aep.md.j2
Comment (rambleraptor) [reply aep/general/0203/aep.md.j2:57]: I'll raise it in Friday's meeting.

File: aep/general/0203/aep.md.j2
Comment (rambleraptor) [reply aep/general/0203/aep.md.j2:165]: I'm re-reading this again and you're right. My brain was substituting "must always be true" for "defaults to true"

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [inline aep/general/0203/aep.md.j2:147]:
```suggestion
- It is a field on a resource that a user provides somewhere as input.
```
Removing the "truthy" statement here - the other sub-bullets LGTM.

File: aep/general/0203/aep.md.j2
Comment (toumorokoshi) [reply aep/general/0203/aep.md.j2:57]: filed an issue: https://github.com/aep-dev/aep.dev/issues/181. Let's raise it for discussion, but won't block the adoption of the AEP.

**Review by toumorokoshi — APPROVED** [review]
One last nit, but otherwise LGTM! I'm approving assuming you'll address the one comment, but will hopefully save you the cycle time on approval. 

Thanks!

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:58]:
```suggestion
used to identify the resource. It **must** be attached to the `path` field and
```

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:211]:
```suggestion
Resource paths, the primary identifiers for any compliant resource, are never
```

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:216]:
```suggestion
Furthermore, in mutation requests, like Standard Update, the resource path as
```

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:220]:
```suggestion
the behavior of the resource path field.
```

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:60]:
```suggestion
paths]).
```

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:249]:
```suggestion
[fields representing resource paths]:
```

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:250]:
```suggestion
  ./resource-paths#fields-representing-resource-paths
```

**Review by mkistler — CHANGES_REQUESTED** [review]
Need to change "name" to "path" in a bunch of places (I think I caught them all).

After that I think we should merge this. I think this AEP will require extensive changes to generalize the guidance and then add OAS-specifics, so we should do that in a separate PR.

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:102]:
```suggestion
- The `ttl` field as described in AEP-214.
```

File: aep/general/0203/aep.md.j2
Comment (mkistler) [inline aep/general/0203/aep.md.j2:122]:
```suggestion
AEP-161).
```

**Review by mkistler — APPROVED** [review]
Two more minor fixes, but I'll approve as these are straightforward.

**Pull request merged**
